### ISSUE TITLE
Session state cookies can be marked as essential

### ISSUE THREAD

**Issue by onovotny** [aspnet/Docs#10443]
The docs here seem to indicate that Session state cookies are not essential but don't say that they can be. 

https://docs.microsoft.com/en-us/aspnet/core/security/gdpr?view=aspnetcore-2.2#tempdata-provider-and-session-state-cookies-are-not-essential

This leads to the assumption that they cannot be marked as essential. Please update the docs and provide a code snippet that shows how to mark it as essential.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 28d0490e-205f-20a5-ace5-904f4c8ad78f
* Version Independent ID: 4a2c8d95-fdd2-fdb1-b61d-802d0a8539c3
* Content: [General Data Protection Regulation (GDPR) support in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/gdpr?view=aspnetcore-2.2#tempdata-provider-and-session-state-cookies-are-not-essential)
* Content Source: [aspnetcore/security/gdpr.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/security/gdpr.md)
* Product: **aspnet-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Tratcher** [comment]
Correct, session can be marked as essential via its options, it's just not by default.

**Comment by Rick-Anderson** [comment]
@Tratcher 	can you give me code to make them essential?

**Comment by guardrex** [comment]
```csharp
services.AddSession(options =>
{
    options.Cookie.IsEssential = true;
});
```

There's a bullet point in the [Session and app state topic (Session state section)](https://docs.microsoft.com/aspnet/core/fundamentals/app-state#session-state) that's a candidate for a note/example about this, too. However, it does link over to this topic, so it might ok as-is.

**Comment by Rick-Anderson** [comment]
This has been fixed

**Comment by Rick-Anderson** [comment]
Fixed in #11306
